### ISSUE TITLE
Redundant let is being fired with different types of let

### ISSUE THREAD

**Issue by wilkerlucio** [borkdude/clj-kondo#901]
**version**

`clj-kondo v2020.05.09`

**platform**

Native

**problem**

Consider the following code:

```clojure
(let [init-link-events (mapv (fn [cid]
                               [::command-init-link
                                {::client-id cid}]) (range clients))]
  (gen/let [chain-size    (gen-chain-size env)
            command-chain (gen-command (-> env
                                           (assoc ::chain-size chain-size)
                                           #_(update ::command-pool into init-link-events)))]
    (into init-link-events (chain->list command-chain))))
```

Kondo thinks this is a redundant let, but they are different types of let that can't be combined.

**expected behavior**

Kondo should not report this is a redundant let when the types of let are different. And I think we can detect it by different symbols.

**Comment by SevereOverfl0w** [comment]
Just to note (based on the slack conversation), there's two possibilities here:

1. As mentioned in the issue, make sure that two of the same kind of let don't match
2. Only enforce this rule for clojure.core/let

I think 1 is a good idea.

**Comment by borkdude** [comment]
@SevereOverfl0w I'm inclined to go with 1 as I know nested core lets are redundant, but could there be macros that are syntactically like let, but for which it make sense to directly nest them? There might be?

**Comment by SevereOverfl0w** [comment]
There's a lot of let-like macros that would benefit from this too.

**Comment by wilkerlucio** [comment]
@borkdude I can imagine the case you describe on 2, sometimes the macro implementation is not let-complete in the sense it can allow only one binding, and in this case the user may need to nest to make multiple usages.

**Comment by wilkerlucio** [comment]
Quick tip if anyone is facing the same issue, a quick hack to get around the validation is:

```clojure
(let [out-binding 0
  out-binding ; write this here, so the validation wont be triggered
  (gen/let [inner-binding (gen/return 1)]
    (+ out-binding inner-binding)))
```

**Comment by borkdude** [comment]
The error was caused not by `:lint-as` config but by the redundant-let linter not looking at the namespace of the let call. This was a bug and is fixed. Nested non-clojure lets were never reported as redundant before, so there isn't a regression. Instead, the idea of letting custom lets partaking in the redundant let linting proposed by @SevereOverfl0w should be discussed an a separate issue.
